fusor/models.py

```diff
@@ -520,18 +547,50 @@ def enforce_abc(cls, values):
         return values
 
     @root_validator(pre=True)
-    def enforce_elements_length(cls, values):
-        """Ensure minimum # of elements."""
-        error_msg = (
+    def enforce_element_quantities(cls, values):
+        """Ensure minimum # of elements, and require > 1 unique genes.
+
+        To validate the unique genes rule, we extract gene IDs from the elements that
+        designate genes, and take the number of total elements. If there is only one
+        unique gene ID, and there are no non-gene-defining elements (such as
+        an unknown partner), then we raise an error.
+        """
+        qt_error_msg = (
             "Fusions must contain >= 2 structural elements, or >=1 structural element "
             "and a regulatory element"
         )
         structural_elements = values.get("structural_elements", [])
         if not structural_elements:
-            raise ValueError(error_msg)
+            raise ValueError(qt_error_msg)
         num_structural_elements = len(structural_elements)
-        if (num_structural_elements + bool(values.get("regulatory_element"))) < 2:
-            raise ValueError(error_msg)
+        reg_element = values.get("regulatory_element")
+        if (num_structural_elements + bool(reg_element)) < 2:
+            raise ValueError(qt_error_msg)
+
+        uq_gene_msg = "Fusions must form a chimeric transcript from two or more genes, or a novel interaction between a rearranged regulatory element with the expressed product of a partner gene."  # noqa: E501
+        gene_ids = []
+        if reg_element:
+            gene_descriptor = cls._access_object_attr(reg_element, "associated_gene")
+            if gene_descriptor:
+                gene = cls._access_object_attr(gene_descriptor, "gene")
+                if gene:
+                    gene_id = cls._access_object_attr(gene, "gene_id")
+                    if gene_id:
+                        gene_ids.append(gene_id)
+
+        for element in structural_elements:
+            gene_descriptor = cls._access_object_attr(element, "gene_descriptor")
+            if gene_descriptor:
+                gene = cls._access_object_attr(gene_descriptor, "gene")
+                if gene:
+                    gene_id = cls._access_object_attr(gene, "gene_id")
+                    gene_ids.append(gene_id)
```

Comment:
This could prob be a function since it is similar. If you don't want to do that, why did you do a check for reg_element if gene_id exists before adding to list but not for the structural elements?

Reply:
Ope that's bug